### DISCUSSION TITLE
Building a Reqnroll project with NUnit on macOS ARM64 architecture

### DISCUSSION THREAD

**Discussion by bgoggan** [reqnroll/Reqnroll#143] [Q&A]
Has anyone successfully build their projects using Reqnroll.Nunit on an ARM64 based architecture? We just recently migrated from Specflow and everything works fine on our testers with windows machines but we have 1 with a Mac based on the above architecture and we receive the below error when building. Any thoughts??

Reqnroll.NUnit.targets(34,5): Error MSB4216 : Could not run the "ReplaceTokenInFileTask" task because MSBuild could not create or connect to a task host with runtime "CLR4" and architecture "x64".  Please ensure that (1) the requested runtime and/or architecture are available on the machine, and (2) that the required executable "/opt/homebrew/Cellar/dotnet@6/6.0.125/libexec/sdk/6.0.125/MSBuild.exe" exists and can be run.

Mac of course doesn't support the .exe extension so I'm not sure how to move forward, if we can.

**Comment by gasparnagy** [comment]
@bgoggan Just that I understand fully... Why the tester with the Mac would need to build Reqnroll itself? Are they would like to contribute to the project?

(Our build runs on Linux and it works there, so the platform itself should not be an issue in general.)

**Reply by bgoggan** [reply]
Sorry, we're trying to get his local environment with our test project built. That's just the error we get once we migrated our code based to Reqnroll from Specflow. We removed all the specflow packages and the specflow extension. Installed all the Reqnroll versions and tried to build our project. Everything worked on the windows machines but when we tried to build his we got that error.

**Reply by gasparnagy** [reply]
@bgoggan Good to know. I have no idea why we need the `ReplaceTokenInFileTask` in that situation, but I will check.

**Reply by gasparnagy** [reply]
@bgoggan And one more question: did you try the v2.0.0 we released yesterday? If not, could you please give it a try on that machine?

**Reply by bgoggan** [reply]
Thank you. I appreciate it.

**Reply by bgoggan** [reply]
Yes, we used all the latest versions, 2.0.0. I just did the migration yesterday.

**Reply by gasparnagy** [reply]
@bgoggan Could you please try to set the environment variable `DISABLEOUTOFPROCTASKHOST` to `1` and retry? (I have found that in https://stackoverflow.com/questions/64553157/what-does-disableoutofproctaskhost-do)

And one more: Could you please try with Reqnroll 1.0.1 as well?

Which SpecFlow version did you use before?

**Reply by bgoggan** [reply]
Ok, we can try that with him. And we were using 3.9.74. I know Reqnroll is built off v4 of Specflow but once I found out it was no longer being maintained I just wanted to jump straight to Reqnroll, which I also know wasn't the suggested path...

**Reply by gasparnagy** [reply]
@bgoggan no, this is absolutely fine, I was just asking to be able to narrow down the problem. Let's see what your investigation finds...

**Reply by bgoggan** [reply]
@gasparnagy Just wanted to let you know I'm hoping to get to try those solutions out today. I appreciate your quick response to this and I'll keep you posted.

**Reply by gasparnagy** [reply]
@bgoggan Do you have any update on this?

**Reply by bgoggan** [reply]
@gasparnagy Good morning, sorry, not yet. We were all gone for the long weekend and he got caught in a production issue yesterday so was unable to try. I'm hoping we can get some time today.

**Reply by bgoggan** [reply]
@gasparnagy We're still getting that same build error after taking the suggestion from StackOverflow by adding the below to the project file.

`
<PropertyGroup>   
     <GenerateResourceMSBuildArchitecture>CurrentArchitecture</GenerateResourceMSBuildArchitecture>
     <GenerateResourceMSBuildRuntime>CurrentRuntime</GenerateResourceMSBuildRuntime>
</PropertyGroup>`

**Reply by bgoggan** [reply]
And I'm not sure if it matters but he's using the Rider IDE. I saw the extension was available so figured that was ok now. We tried VS code as well with the same results

**Reply by gasparnagy** [reply]
@bgoggan Did you also try to set the environment variable `DISABLEOUTOFPROCTASKHOST` to `1`?

Could you also try simply from the console: `dotnet build`?

Could you please paste here the project file?

**Reply by gasparnagy** [reply]
There is also a suggestion here: https://youtrack.jetbrains.com/issue/RIDER-46357/Build-error-MSB4216-depending-on-Rider-version-and-target-architecture

**Reply by gasparnagy** [reply]
And also here: https://rider-support.jetbrains.com/hc/en-us/community/posts/7461118319122-weird-error-in-rider-happened-from-nowhere

**Reply by bgoggan** [reply]
Yes, we tried with the dotnet build command and I'll look at those other suggestions but the reporters of those issues appear to be on a Windows OS which is not our case.

**Reply by gasparnagy** [reply]
@bgoggan Thx. Yeah... Sorry for flooding you with random ideas. I am really struggling to find out what could cause this. All of our code is AnyCPU .NET 4.6.2 or .NET Standard 2.0, so it should not cause such an issue. And all reports I found around that suggested that it was more or an installation problem, not a problem with the task itself. (Of course it does not mean that it is not.)

Maybe to narrow the issue, you can try to download the [Renqroll quickstart sample](https://github.com/reqnroll/Quickstart) and try to build it on that machine.

**Reply by gasparnagy** [reply]
I was able to reproduce it on a friend's MacBook with the quickstart sample... It works with Reqnroll 1.0.0, so it is a regression issue. I try to figure out what is wrong.

**Reply by bgoggan** [reply]
Ok, I hadn't tried downgrading to the previous version yet. We can do that for now if needed, assuming we don't lose anything feature-wise doing that.

**Reply by gasparnagy** [reply]
See related issue: https://github.com/reqnroll/Reqnroll/issues/152

**Reply by bgoggan** [reply]
Great, thank you!

**Reply by gasparnagy** [reply]
@bgoggan no need to downgrade. I will make a release with the fix tomorrow.

**Reply by bgoggan** [reply]
Thank you for the quick solution here. We'll keep an eye out for the new version.

**Reply by gasparnagy** [reply]
v2.0.2 is out!

**Reply by bgoggan** [reply]
Great, let me update our packages and we'll let you know how it goes.

**Reply by bgoggan** [reply]
@gasparnagy We are now able to build the projects but the .feature files are not created, so all we had was the .feature.cs files.....We're going to completely uninstall his version of Rider and start fresh just in case something was wrong. We'll start again with the QuickStart project and see what happens. I'll keep you posted.

**Reply by ajeckmans** [reply]
The feature files are there. It is a bug in the rider plugin which hides the feature files (or well, excludes them from the project visually). You can use the workaround posted here: https://github.com/reqnroll/Reqnroll.Rider/issues/1

**Reply by bgoggan** [reply]
@ajeckmans So, the workaround is to add the new itemgroup with the <Content Include="Features\*.feature"/> value? We tried that but still the features do not appear in the solution.

**Reply by bgoggan** [reply]
@ajeckmans Sorry, standby. It worked on the QuickStart project so we're looking more closely at our project.

**Reply by bgoggan** [reply]
@ajeckmans Ok, we had to update our workaround to the below since we nest our features in sub-directories but it does work.
`<ItemGroup>
  <Content Include="Features\**\*.feature"/>
</ItemGroup>`
